internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -955,6 +955,13 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 		dirty = true
 	}
 
+	// Disable AutomatedCleaningMode if the converged flow is not enabled
+	// since AutomatedCleaning requires IPA, but disabling the converged flow
+	// disables IPA.
+	if !r.ConvergedFlowEnabled && bmh.Spec.AutomatedCleaningMode != bmh_v1alpha1.CleaningModeDisabled {
```

Comment:
I don't think we need to check ConvergedFlowEnabled here because we will have always returned [here](https://github.com/openshift/assisted-service/pull/6662/files#diff-86415614c6ed40c56c5ca2b5f751a65746ef5c83b61e6f021a4f5d817e4425baL941)

Also we should set `dirty = true` in this condition

Reply:
Ah thank you! Modified